docs/diagrams/kubernetes_workflow.md

```diff
@@ -0,0 +1,1155 @@
+    style HandleNotFound fill:#ff6b35
+    style FailStdin fill:#ff6b35
+    style FailContext fill:#ff6b35
+    style ExitContext fill:#4a9eff
+    style Success fill:#4a9eff
+    style SuccessNormal fill:#4a9eff
+    style FailInterrupted fill:#ff6b35
+    style FailNonEOF fill:#ff6b35
+    style FailAuth fill:#ff6b35
+    style FailTimeout fill:#ff6b35
+```
+
+## Key Features
+
+### Resilience Mechanisms
+
+1. **Automatic Reconnection**: Logger method automatically reconnects on stream disconnection
+2. **Retry Logic**: Fibonacci backoff (exponential-like) for transient errors, with no retry limit for EOF with Running state
+3. **Duplicate Detection**: Timestamp-based log line deduplication during reconnections
+4. **Timeout Handling**: Configurable timeouts for pod pending state
+5. **Graceful Degradation**: Falls back to no-reconnect method for older Kubernetes versions
+6. **Long-Running Job Support**: Continues attempting reconnection indefinitely when EOF occurs with Running containers (handles 4-hour log stream timeouts)
+
+### Configuration Flexibility
+
+1. **Multiple Auth Methods**: kubeconfig, incluster, or runtime
+2. **Runtime Overrides**: Allow dynamic image/command/params/pod specification
+3. **Stream Method Selection**: Choose logger or TCP method
+4. **Rate Limiting**: Configurable QPS and burst for API calls
+5. **Timeout Configuration**: Environment variables for timeouts and retry counts
+
+## Configuration Options
+
+### Environment Variables
+
+- `RECEPTOR_KUBE_TIMEOUT_START`: Base timeout for retries (default: 1s, max: 1m)
+- `RECEPTOR_KUBE_RETRY_COUNT`: Number of retries (default: 5, max: 100)
+- `RECEPTOR_KUBE_SUPPORT_RECONNECT`: Enable/disable/auto reconnect (default: enabled)
+- `RECEPTOR_KUBE_CLIENTSET_QPS`: API rate limit QPS (default: 100)
+- `RECEPTOR_KUBE_CLIENTSET_BURST`: API rate limit burst (default: 10x QPS)
+- `RECEPTOR_KUBE_CLIENTSET_RATE_LIMITER`: Rate limiter type (never/always/tokenbucket)
+
+### Worker Configuration
+
+```yaml
+- work-kubernetes:
+    workType: k8s-worker
+    namespace: default
+    image: my-image:latest
+    command: /bin/sh
+    params: -c
+    authMethod: incluster
+    streamMethod: logger
+    allowRuntimeCommand: false
+    allowRuntimeParams: true
+    deletePodOnRestart: true
+```
+
+## Work States
+
+1. **WorkStatePending (0)**: Initial state, connecting to Kubernetes or creating pod
+2. **WorkStateRunning (1)**: Pod running, streaming data
+3. **WorkStateSucceeded (2)**: Work completed successfully. Determined by:
+   - Exit code 0, OR
+   - Exit code != 0 but termination reason is "Completed" or "Error" (indicates normal program completion with error)
+4. **WorkStateFailed (3)**: Work failed. Determined by:
+   - Exit code != 0 AND termination reason indicates interruption (OOMKilled, Evicted, etc.), OR
+   - Other errors occurred during execution (stream errors, pod failures, etc.)
+5. **WorkStateCanceled (4)**: Work was canceled, pod deleted
```

Comment:
Are work states specific to kubernetes execution or are these global work state definitions that also apply to command?

Put another way, if we wanted to add more work states to represent different stages of `Pending` in a Kubernetes execution, would we break things?